Add secondary, versioned configuration schema for readability and flexibility.

pcs3rd opened the pull request:
I'm hoping that breaking configuration options out into smaller subjections helps to maintain readability.  
This also proposes a version identifier to help ensure backwards compatibility.

SpudGunMan commented:
I'm kinda lost, there is no backwards compatibility as any missing values are defaulted? 

I don't expect any enhancements that would impact the future?

pcs3rd commented:
I renamed the branch and moved commits to a feature branch, but I am intending to add another module for scavenger hunt-type operations and possibly checkpoints.   
  
This makes me familiar with how meshing around is configured, while attempting to make the config file more readable, while not breaking previous configurations.

SpudGunMan commented:
Ahh got it, set defaults to what most would use.  

Did you see the unfinished similar game code in the modules folder

SpudGunMan commented:
Also I have purposely not used regex
